### PR TITLE
Move JsonPatch to Azure.Core

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/api/Azure.Core.net461.cs [hunk 1]
```diff
@@ -55,6 +55,24 @@ public partial class AzureKeyCredential
         public static bool operator !=(Azure.HttpRange left, Azure.HttpRange right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class JsonPatchDocument
+    {
+        public JsonPatchDocument() { }
+        public JsonPatchDocument(Azure.Core.Serialization.ObjectSerializer serializer) { }
+        public JsonPatchDocument(System.ReadOnlyMemory<byte> rawDocument) { }
```
Comment (pakrym) [inline sdk/core/Azure.Core/api/Azure.Core.net461.cs:62]: @alzimmermsft FYI, we've added a way to pass an existing document as `JsonPatchDocument` with no overhead.

File: sdk/core/Azure.Core.Experimental/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## 0.1.0-preview.7 (Unreleased)
 
+### Breaking Changes
+- `JsonPatchDocument` type moved to `Azure.Core` package.
```
Comment (heaths) [inline sdk/core/Azure.Core.Experimental/CHANGELOG.md:6]: Maybe we don't care enough about the Expirmental package, but you could also use the TypeForwardToAttribute like the BCL has done. In that case, this isn't breaking, per se, but still worth mentioning.

File: sdk/core/Azure.Core/src/JsonPatchDocument.cs [hunk 3]
```diff
@@ -4,41 +4,61 @@
...
+        public JsonPatchDocument(ReadOnlyMemory<byte> rawDocument, ObjectSerializer serializer)
         {
             _operations = new Collection<JsonPatchOperation>();
+            _rawDocument = rawDocument;
             _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
         }
 
         /// <summary>
-        /// Appends an "add" operation to this <see cref="JsonPatchDocument"/>.
+        /// Initializes a new instance of <see cref="JsonPatchDocument"/> using an existing UTF8 encoded JSON Patch document.
```
Comment (heaths) [inline sdk/core/Azure.Core/src/JsonPatchDocument.cs:61]: Nit:
```suggestion
        /// Initializes a new instance of <see cref="JsonPatchDocument"/> using an existing UTF8-encoded JSON Patch document.
```

File: sdk/core/Azure.Core/src/JsonPatchDocument.cs [hunk 4]
```diff
@@ -126,28 +146,53 @@ public void AppendTest<T>(string path, T value)
         }
 
         /// <summary>
-        /// Returns a formatted JSON string representation of this <see cref="JsonPatchDocument"/>.
+        /// Returns a UTF8 encoded representation of this <see cref="JsonPatchDocument"/> instance.
```
Comment (heaths) [inline sdk/core/Azure.Core/src/JsonPatchDocument.cs:149]: Nit:
```suggestion
        /// Returns a UTF8-encoded representation of this <see cref="JsonPatchDocument"/> instance.
```

Comment (heaths) [reply sdk/core/Azure.Core/src/JsonPatchDocument.cs:149]: A few others below. I'll not repeat. Basically, it's a compound adjective and should be hyphenated, but not a big deal (i.e. nit).

**Review by heaths — APPROVED** [review]

File: sdk/core/Azure.Core.Experimental/CHANGELOG.md (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/CHANGELOG.md:6]: Yeah, we don't care enough about this in Experimental. And in the case of this type we've changed the namespace which AFAIK doesn't allow us to use TFTA.

**Pull request merged**